wire-ios-request-strategy/Sources/E2EIdentity/CRL/CRLsDistributionPointsObserver.swift

```diff
@@ -0,0 +1,55 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import Combine
+import WireDataModel
+
+public protocol CRLsDistributionPointsObserving {
+
+    func startObservingNewCRLsDistributionPoints(
+        from publisher: AnyPublisher<CRLsDistributionPoints, Never>
+    )
+
+}
+
+public class CRLsDistributionPointsObserver: CRLsDistributionPointsObserving {
+    private var cancellables: Set<AnyCancellable> = []
+    private let cRLsChecker: CertificateRevocationListsChecking
+
+    public init(cRLsChecker: CertificateRevocationListsChecking) {
+        self.cRLsChecker = cRLsChecker
+    }
+
+    // Is this necessary?
+    deinit {
+        cancellables.forEach { $0.cancel() }
+    }
+
+    public func startObservingNewCRLsDistributionPoints(
+        from publisher: AnyPublisher<CRLsDistributionPoints, Never>
+    ) {
+        publisher.sink { distributionPoints in
+            Task {
+                await self.cRLsChecker.checkNewCRLs(from: distributionPoints)
+            }
+        }
+        .store(in: &cancellables)
+    }
```

Comment:
I'm don't get the picture how the system components are wired. Can CRLsChecker not just listen to a `NotificationCenter` notification instead of passing combine publishers through different layers and classes? Maybe I miss some points..

For example I ask myself here: what happens with the result of checked new CRLS?

Reply:
created a ticket to improve this later https://wearezeta.atlassian.net/browse/WPB-6673